core/src/main/java/org/apache/accumulo/core/data/RowRange.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.core.data;
+
+import org.apache.hadoop.io.Text;
+
+/**
+ * This class is used to specify a range of rows.
+ */
+public class RowRange {
+  final private Text startRow;
+  final private Text endRow;
+  final private boolean startRowInclusive;
+  final private boolean endRowInclusive;
+  private final boolean infiniteStartRow;
+  private final boolean infiniteEndRow;
+
+  /**
+   * Creates a range of rows that goes from negative to positive infinity
+   */
+  public RowRange() {
+    this(null, true, null, true);
+  }
+
+  /**
+   * Creates a range of rows from startRow inclusive to endRow inclusive.
+   *
+   * @param startRow starting row; set to null for negative infinity
```

Comment:
```suggestion
   * @param startRow starting row; set to null for the smallest possible row (an empty one)
```

Instead of overloading constructors, you could have static entry points to more clearly express bounded and unbounded ranges. See [Guava's Range API](https://guava.dev/releases/19.0/api/docs/com/google/common/collect/Range.html) for a really good example of static entry points with phrases like "atLeast(lowerBound)", or "atMost(upperBound)" or "between(lowerBound,upperBound)". If we're adding any new Range API, these would be really useful to have right up front, rather than trying to add them in later.

Reply:
Addressed the javadoc fix in 46db626

Reply:
Added the static entry points in 2d63f6f